OutOfSchool/OutOfSchool.WebApi.Tests/Services/ProviderServicesTests/ProviderServiceV2Tests.cs

```diff
@@ -0,0 +1,580 @@
+        var localizer = new Mock<IStringLocalizer<SharedResource>>();
+        var logger = new Mock<ILogger<ProviderServiceV2>>();
+        var workshopServicesCombiner = new Mock<IWorkshopServicesCombiner>();
+        var providerImagesService = new Mock<IImageDependentEntityImagesInteractionService<Provider>>();
+        var changesLogService = new Mock<IChangesLogService>();
+        notificationService = new Mock<INotificationService>(MockBehavior.Strict);
+        providerAdminService = new Mock<IProviderAdminService>();
+        institutionAdminRepositoryMock = new Mock<IInstitutionAdminRepository>();
+        currentUserServiceMock = new Mock<ICurrentUserService>();
+        ministryAdminServiceMock = new Mock<IMinistryAdminService>();
+        regionAdminServiceMock = new Mock<IRegionAdminService>();
+        codeficatorServiceMock = new Mock<ICodeficatorService>();
+        regionAdminRepositoryMock = new Mock<IRegionAdminRepository>();
+        averageRatingServiceMock = new Mock<IAverageRatingService>();
+        areaAdminServiceMock = new Mock<IAreaAdminService>();
+        areaAdminRepositoryMock = new Mock<IAreaAdminRepository>();
+        userServiceMock = new Mock<IUserService>();
+        communicationService = new Mock<ICommunicationService>();
+
+        var authorizationServerConfig = Options.Create(new AuthorizationServerConfig { Authority = new Uri("http://test.com") });
+        mapper = TestHelper.CreateMapperInstanceOfProfileType<MappingProfile>();
+
+        providerService = new ProviderServiceV2(
+            providersRepositoryMock.Object,
+            usersRepositoryMock.Object,
+            logger.Object,
+            localizer.Object,
+            mapper,
+            addressRepo.Object,
+            workshopServicesCombiner.Object,
+            providerAdminRepositoryMock.Object,
+            providerImagesService.Object,
+            changesLogService.Object,
+            notificationService.Object,
+            providerAdminService.Object,
+            institutionAdminRepositoryMock.Object,
+            currentUserServiceMock.Object,
+            ministryAdminServiceMock.Object,
+            regionAdminServiceMock.Object,
+            codeficatorServiceMock.Object,
+            regionAdminRepositoryMock.Object,
+            averageRatingServiceMock.Object,
+            areaAdminServiceMock.Object,
+            areaAdminRepositoryMock.Object,
+            userServiceMock.Object,
+            authorizationServerConfig,
+            communicationService.Object);
+        providersRepositoryMock.Setup(w => w.UnitOfWork.CompleteAsync()).ReturnsAsync(It.IsAny<int>());
+    }
+
+    #region Create
+
+    [TestCase(null, ProviderLicenseStatus.NotProvided)]
+    [TestCase("1234567890", ProviderLicenseStatus.Pending)]
+    public async Task Create_WhenEntityIsValid_ReturnsCreatedEntity(string license, ProviderLicenseStatus expectedLicenseStatus)
```

Comment:
(Tests completeness) `providerImagesService` methods has to be called after provider creation (via `Create()` method) if there is cover image or image files set. This is not tested.

Same for `Delete()`.